### PR TITLE
chore(cli): enable creating environments from temporary creds

### PR DESCRIPTION
**Pull request by efekarakus**
- Add the struct to hold temporary credentials.
- Validate temp credentials flags.
- Create clients from temporary credentials if specified.

The PR to ask for the credentials, updating `env delete`, and `env init` will be follow-ups.

Related #1228 

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: internal/pkg/cli/env_init.go [hunk 1]
```diff
@@ -48,11 +51,19 @@ var (
...
+
 type initEnvVars struct {
 	*GlobalOpts
-	EnvName      string // Name of the environment.
-	EnvProfile   string // AWS profile used to create an environment.
-	IsProduction bool   // Marks the environment as "production" to create it with additional guardrails.
+	EnvName      string        // Name of the environment.
+	EnvProfile   string        // AWS profile used to create an environment.
+	IsProduction bool          // Marks the environment as "production" to create it with additional guardrails.
+	TempCreds    tempCredsVars // Temporary credentials to initialize the environment. Mutually exclusive with the AWS profile.
```
Comment (SoManyHs) [inline internal/pkg/cli/env_init.go:65]: Should there be something here about only one of `EnvProfile` or `TempCreds` can be used or does it matter?

File: internal/pkg/cli/env_init.go [hunk 2]
```diff
@@ -67,17 +78,33 @@ type initEnvOpts struct {
...
+	var sess *session.Session
+	if o.EnvProfile != "" {
+		profileSess, err := sessions.NewProvider().FromProfile(o.EnvProfile)
+		if err != nil {
+			return fmt.Errorf("create session from profile %s: %w", o.EnvProfile, err)
+		}
+		sess = profileSess
+	} else {
+		staticSess, err := session.NewSession(&aws.Config{
+			Credentials: credentials.NewStaticCredentials(o.TempCreds.AccessKeyID, o.TempCreds.SecretAccessKey, o.TempCreds.SessionToken),
```
Comment (SoManyHs) [inline internal/pkg/cli/env_init.go:95]: Do we need to nil check `o.TempCreds`?

File: internal/pkg/cli/env_init.go [hunk 3]
```diff
@@ -86,8 +113,7 @@ func newInitEnvOpts(vars initEnvVars) (*initEnvOpts, error) {
 	if err != nil {
 		return nil, err
 	}
-	sessProvider := sessions.NewProvider()
-	defaultSession, err := sessProvider.Default()
+	defaultSession, err := sessions.NewProvider().Default()
```
Comment (SoManyHs) [inline internal/pkg/cli/env_init.go:116]: Do we not need to instantiate `tempCredsVars` here to avoid NPEs downstream?

File: internal/pkg/cli/env_init_test.go [hunk 4]
```diff
@@ -52,6 +80,12 @@ func TestInitEnvOpts_Validate(t *testing.T) {
 				initEnvVars: initEnvVars{
 					EnvName:    tc.inEnvName,
 					GlobalOpts: &GlobalOpts{appName: tc.inAppName},
+					EnvProfile: tc.inProfileName,
+					TempCreds: tempCredsVars{
```
Comment (SoManyHs) [inline internal/pkg/cli/env_init_test.go:84]: Can we add a test if this is nil?

File: internal/pkg/cli/env_init.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/cli/env_init.go:65]: There is a comment for TempCreds: `Mutually exclusive with the AWS profile.` and `Validate()` will throw an error if both are set. 

Or am I misunderstanding/missing something?

File: internal/pkg/cli/env_init.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/cli/env_init.go:95]: One of `TempCreds` or `EnvProfile` will be initialized after the `Ask()` method but I haven't implemented that in this PR 😋

Right now this PR doesn't surface these flags yet to the customer, the behavior is exactly the same as before it justs enables the new codepath during `Execute`.

File: internal/pkg/cli/env_init_test.go (hunk 4 above)
Comment (efekarakus) [reply internal/pkg/cli/env_init_test.go:84]: `TempCreds`'s zero value is the empty `tempCredsVars{}` struct which sets all these credential values to the empty string ""

File: internal/pkg/cli/env_init.go (hunk 1 above)
Comment (SoManyHs) [reply internal/pkg/cli/env_init.go:65]: Bah i meant to put this comment on line 63. I think elsewhere our phrasing has typically been "only one of X or Y can be used" but the mutually exclusive thing in this line is fine.

File: internal/pkg/cli/env_init.go (hunk 2 above)
Comment (SoManyHs) [reply internal/pkg/cli/env_init.go:95]: Ooohh. I think that was what I was missing -- I was going to mention above in the description of the options fields that referring to the `EnvProfile` as `AWS Profile` might be confusing but that I assumed the flag description would use the actual flag name.

Comment (efekarakus) [reply internal/pkg/cli/env_init.go:137]: Added comments to "Profile" as well saying that it's mutually exclusive with "TempCreds"
